src/Configuration/Config.Json/src/JsonConfigurationFileParser.cs

```diff
@@ -19,89 +17,70 @@ internal class JsonConfigurationFileParser
         private readonly Stack<string> _context = new Stack<string>();
         private string _currentPath;
 
-        private JsonTextReader _reader;
-
         public static IDictionary<string, string> Parse(Stream input)
             => new JsonConfigurationFileParser().ParseStream(input);
 
         private IDictionary<string, string> ParseStream(Stream input)
         {
             _data.Clear();
-            _reader = new JsonTextReader(new StreamReader(input));
-            _reader.DateParseHandling = DateParseHandling.None;
-
-            var jsonConfig = JObject.Load(_reader);
 
-            VisitJObject(jsonConfig);
+            using (var reader = new StreamReader(input))
+            using (JsonDocument doc = JsonDocument.Parse(reader.ReadToEnd(),
+                new JsonReaderOptions { CommentHandling = JsonCommentHandling.Skip }))
+            {
+                VisitElement(doc.RootElement);
```

Comment:
What if the root element is not of type object (for example array or single value)? Is it fine that we would just throw on enumerate object since such an input is unexpected?

Reply:
Yeah we would have thrown an exception before as well, so this would just be a different exception